platform/posix/transport/CMakeLists.txt

```diff
@@ -6,7 +6,7 @@ include( ${PLATFORM_DIR}/posix/posixFilePaths.cmake )
 if( NOT EXISTS ${MODULES_DIR}/standard/coreMQTT/source )
     # Attempt to clone coreMQTT.
     if( ${BUILD_CLONE_SUBMODULES} )
```

Comment:
Is this logic redundant considering that if the `BUILD_CLONE_SUBMODULES` flag is set, it will automatically clone all submodules?

Reply:
I think if the `CMakeLists.txt` file from the platform directory is included, then transport_interface.h needs to be there for platform to build. Ideally, it would be best to keep platform independent from libraries, but this is another way around that. In this case, if the submodule is already cloned, nothing will happen.